XenAdmin/Wizards/PatchingWizard/PatchingWizard_UploadPage.cs

```diff
@@ -509,6 +509,23 @@ private void singleAction_Completed(ActionBase sender)
                                 AllIntroducedPoolUpdates.Add(PoolUpdate, SelectedNewPatchPath);
                             }
                         }
+
+                        var uploadAction = action as UploadSupplementalPackAction;
+                        if (uploadAction != null)

```

Comment:
`uploadAction != null` is always true here, because of the check on line 489.
We should move these two lines to replace the check on line 489 and use the new variable, uploadAction , everywhere inside the `if` block.